File: programmers_summer/q2.py


def solution(t, r):
    answer = []

    ids=[i for i in range(len(t))]
    # id,lift_num,rank(smaller better)
    persons=[]
    for i in range(len(ids)):
        persons.append([ids[i],t[i],r[i]])
    # persons=list(zip(ids,t,r))
    while persons:
        persons.sort(key=lambda x: (x[1], x[2], x[0]))
        print(persons)
        person=persons.pop(0)
        answer.append(person[0])
        for p in persons:
            if p[1]==person[1]:
                p[1]+=1




    return answer



t=[7,6,8,1]
r=[0,1,2,3]
print(solution(t,r))

File: ProgrammersKitJS/q2.js
function solution(deposit) {
  let answer = [];
  for(let d of deposit){
    if(d>0){
      answer.push(d);
    }else{
      let rest=d;
      while(rest<0){
        let money=answer.pop();
        rest+=money;
      }
      if(rest!=0){
        answer.push(rest);
      }
    }
  }
  return answer;
}

let deposit=[500, 1000, 2000, -1000, -1500, 500]	;
console.log(solution(deposit));